Auth > API 필드 유효성 정책 동기화를 위한 정적파일 추가

wch-os opened the pull request:
## Issues

- Resolves #445 


## Description

- Auth API에서 사용되는 필드에 대한 유효성 검증 규칙을 담은 정적 파일 구현
  - 유효성 정책 정의 필드: 이메일, 비밀번호, 비밀번호 확인, 사용자명 

- 각 필드의 정규표현식, 길이, 메시지 정의

<br />

## Review Points

- 자유로운 리뷰 부탁드립니다.

<br />

## How Has This Been Tested?

- 포스트맨으로 정의한 정적 파일이 정상 응답되고 있음을 확인했습니다.

<br />

wch-os committed:
feat: add auth validation rules

- 이메일, 비밀번호, 비밀번호 확인, 사용자명 필드에 대한 유효성 검증 규칙 추가
- 각 필드의 정규표현식, 길이, 메시지 정의

wch-os committed:
feat(monolith): exposes auth validation rules endpoint in jwt filter

- 유효성 규칙 엔드포인트를 jwt 인가 없이 접근할 수 있도록 규칙 추가
- 엔드포인트를 auth 도메인에 맞추도록, validation-rules 파일 위치 수정

merge-simpson commented on services/auth/api/domain/src/main/resources/validation/auth-rules.json:
### **💬 (On Discussion) 정적 파일 위치**

어느 모듈에 두는 게 좋을지 논의 중입니다!

merge-simpson replied:
다음처럼 결정하였으니, 참고 부탁드립니다!

---

- Domain 모듈에 두기  
  단, 커스텀 정적 경로를 만들어 제공 (+ 실행 모듈에 정적배포 설정을 추가하여 제어)

> 배포 전략의 수정이 도메인 모듈에 영향을 주지 않기 위함.

merge-simpson reviewed (commented):
@wch-os 님, 언제나 리즈너블한 제안과 작업 감사합니다.

이번 리뷰는 논의 후 다시 이어 가겠습니다!

merge-simpson reviewed (commented):
수정하시면 다시 리뷰하겠습니다...!

wch-os committed:
refactor(auth): moves validation rules to domain module.

- 유효성 검증 규칙 파일은 도메인에 대한 유효성 정책이므로 domain 모듈 내에 위치하도록 수정

- 유효성 검증 규칙 파일을 배포 과정에서 domain 모듈의 추가 수정 없이 배포 전략을 유연하게 바꿀 수 있도록, /resources/validation 커스텀한 경로에 위치하도록 함

wch-os committed:
feat(monolith): configure static resource handling for validation rules

- /validation 으로 시작하는 모든 요청은 정적 파일을 조회하는 요청으로 간주하고, src/main/resources/validation의 커스텀한 경로 안에서 정적 파일을 조회해서 응답하도록 함

wch-os committed:
feat(monolith): permits access to validation files

- jwt 액세스 토큰 없이 유효성 검증 파일에 접근할 수 있도록 규칙 추가

wowddok99 reviewed (approved):
작성해주신 코드 확인하였습니다.
병합해주셔도 될듯합니다 : ) 
작업 고생하셨어요!
